accessors/datastream/datastream_accessor.go

```diff
@@ -0,0 +1,124 @@
+}
+type DatastreamAccessorImpl struct {}
+
+
+// FetchTargetBucketAndPath fetches the bucket and path name from a Datastream destination config.
+func (da *DatastreamAccessorImpl) FetchTargetBucketAndPath(ctx context.Context, datastreamClient datastreamclient.DatastreamClient, projectID string, datastreamDestinationConnCfg streaming.DstConnCfg) (string, string, error) {
+	if datastreamClient == nil {
+		return "", "", fmt.Errorf("datastream client could not be created")
+	}
+	dstProf := fmt.Sprintf("projects/%s/locations/%s/connectionProfiles/%s", projectID, datastreamDestinationConnCfg.Location, datastreamDestinationConnCfg.Name)
+	// `GetConnectionProfile` has out of box retries. Ref - https://github.com/googleapis/googleapis/blob/master/google/cloud/datastream/v1/datastream_grpc_service_config.json
+	res, err := datastreamClient.GetConnectionProfile(ctx, dstProf)
+	if err != nil {
+		return "", "", fmt.Errorf("could not get connection profiles: %v", err)
+	}
+	// Fetch the GCS path from the target connection profile.
+	// The Get calls for Google Cloud Storage API have out of box retries.
+	// Reference - https://cloud.google.com/storage/docs/retry-strategy#idempotency-operations
+	gcsProfile := res.Profile.(*datastreampb.ConnectionProfile_GcsProfile).GcsProfile
+	bucketName := gcsProfile.Bucket
+	prefix := gcsProfile.RootPath + datastreamDestinationConnCfg.Prefix
+	prefix = utils.ConcatDirectoryPath(prefix, "data/")
+	return bucketName, prefix, nil
+}
+
+// Deletes a connection Profile
+func (da *DatastreamAccessorImpl) DeleteConnectionProfile(ctx context.Context, datastreamClient datastreamclient.DatastreamClient, id string, projectId string, region string) error {
+	op, err :=datastreamClient.DeleteConnectionProfile(ctx,  &datastreampb.DeleteConnectionProfileRequest{
+		Name: fmt.Sprintf("projects/%s/locations/%s/connectionProfiles/%s", projectId, region, id),
+	})
+	if err != nil {
+		return err
+	}
+	
+	err = op.Wait(ctx)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Creates new connection Profile
+func (da *DatastreamAccessorImpl) CreateConnectionProfile(ctx context.Context, datastreamClient datastreamclient.DatastreamClient, req *datastreampb.CreateConnectionProfileRequest) (*datastreampb.ConnectionProfile, error) {
+	op, err :=datastreamClient.CreateConnectionProfile(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	return op.Wait(ctx)
+}
+// Gets all connection profiles in a region
+func (da *DatastreamAccessorImpl) GetConnProfilesRegion(ctx context.Context, datastreamClient datastreamclient.DatastreamClient, projectId string, region string) ([]string, error) {
+	profilesIt := datastreamClient.ListConnectionProfiles(ctx, &datastreampb.ListConnectionProfilesRequest{Parent: "projects/" + projectId + "/locations/" + region})
+	var profiles []string = []string{}
+	for {
+		resp, err := profilesIt.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return profiles, err
```

Comment:
Wondering if its a good idea to return whatever profiles we have collected till now. Most uses cases I have seen till now seem to return a nil/empty in case of error. Also keeps the output deterministic in case datastream decides to change orders.